docs(kubernetes) Migrate KIC documentation

lena-larionova opened the pull request:
**Note**: All commits before November 10th are inherited history from the ingress controller repo where these docs originate (https://github.com/Kong/kubernetes-ingress-controller). Please review changes made in the November 10th and onward commits.

Docs issue: https://konghq.atlassian.net/browse/DOCS-1301

### Why is this needed?
This is a big, big customer and user ask. With this migration, the hope is to have all the docs in one place for easy access.

### Changes
Migrating the `/docs` directory from `kong/kubernetes-ingress-controller/main` into `kong/docs.konghq.com`, and completing the minimal necessary tasks to display a useable doc on the live site. Specifically:
* Migrate all content under [`docs`](https://github.com/Kong/kubernetes-ingress-controller/tree/main/docs) into `app/kubernetes-ingress-controller/1.0.x`.
* Create version and edition metadata for `kong-ingress-controller` and `1.0.x`
* Navigation (table of contents) file with icons
* Update all relative links to point to this repo (or in a rare case, back to the originating repo)
* Create a landing page
* Add top-level tab for Kong Ingress Controller to doc site
* Remove extraneous "Table of Contents" sections from docs, as the doc site pulls this information from page metadata to create a ToC already
* Make sure images aren't broken

### Preview
https://deploy-preview-2446--kongdocs.netlify.app/kubernetes-ingress-controller/latest/introduction

### Outstanding work
* The [Plugin Compatibility](https://deploy-preview-2446--kongdocs.netlify.app/kubernetes-ingress-controller/1.0.x/references/plugin-compatibility) and [Version Compatibility](https://deploy-preview-2446--kongdocs.netlify.app/kubernetes-ingress-controller/1.0.x/references/version-compatibility/) topics/tables don't display very well. Either need to convert the icons into something else, or add a style definition for them (or hide them temporarily and revisit). 
  * Do we want to have these topics in these pages, or would you rather maintain them in the original repo @rainest @hbagdi ?
* Didn't touch any content or attempt any cleanup; will look into that in the coming weeks.
* Add the kong/charts and kustomize docs (if any) to this doc set, if possible (way out of scope for this PR, but would be ideal eventually).
* Add a search index (won't work in preview, has to be live before doing that)
* Question for @eskibars and @rezaloo: as with decK, the original plan was to add the docs to a top-level tab on the doc site. After some recent conversations, it sounds like that's probably not the right approach anymore, and instead we want to treat it as any Gateway component. Do I have that right? 
  * Since KIC is an independently-versioned component, it might make sense to treat it like decK as well: no tab, but with its own section of the site: https://docs.konghq.com/deck

mflendrich commented on app/kong-ingress-controller/1.0.x/index.md:
```diff
@@ -0,0 +1,34 @@
+---
+<div class="docs-grid">
+
+  <h2 class="docs-grid-section-title">Foundations</h2>
```
Might be bikeshedding, but I think that we should feature "Guides" here as well. Consider this optional.

lena-larionova replied:
Might as well. 

Actually - if you take a look at the Enterprise landing page (https://docs.konghq.com/enterprise/), there's a section with a couple of the most popular guides and references (or, things that we just want to call out, really). Could you identify 2-4 of the most useful/popular docs that I could link in a section like that?

rainest replied:
IMO using-consumer-credential-resource.md and using-kongplugin-resource.md for sure--they're basic tasks that almost everyone will perform. If we want more, setting-up-custom-plugins.md (fairly common) and cert-manager.md (useful and we get a fair number of questions related to it).

lena-larionova replied:
So - this topic is now gone, as Shane and co don't want KIC at the top level as its own "product". However, I can still carve out a space on the introduction and the guides landing page for some kind of spotlight for important guides.

mflendrich commented on jekyll.yml:
```diff
@@ -108,6 +108,11 @@ defaults:
       layout: 'docs-v2'
 
+  - scope:
+      path: 'kong-ingress-controller'
```
We have some naming inconsistency: "kubernetes ingress controller" vs "kong ingress controller" vs "kong for kubernetes". Is it within the scope of this PR to assume "kong ingress controller" as our choice for all communication?

lena-larionova commented on jekyll-dev.yml:
```diff
@@ -131,3 +136,4 @@ ee_product_name: Kong Enterprise
 ce_product_name: Kong Gateway
 base_gateway: Kong Gateway
 mesh_product_name: Kong Mesh
+kic_product_name: Kong Ingress Controller
```
The PR isn't truly deciding it, but it is going off of recent Product team decisions. I'll double-check with them - I was fairly sure that the name is supposed to be "Kong Ingress Controller" going forward, but I have doubts now.

lena-larionova replied:
You were right, it is officially Kubernetes Ingress Controller. I've set this as the variable name now, and will do a concentrated cleanup effort in a later PR to make sure that variable is used consistently across the doc.

lena-larionova commented:
From Shane's feedback:
* Removed Kong Ingress Controller tab from top nav
* Removed landing page
* Added it to the Enterprise nav as test, and added Enterprise nav entries to KIC nav
  * This is a solution to a problem brought up by Shane around doc flows. Unfortunately, because KIC has its own versioning scheme, it can't easily live under /enterprise or CE docs. My attempted (read: hacky) solution is to populate both the Enterprise and Community navs with the KIC topics.

lena-larionova commented:
Side comment: I really should've merged the migrated content over first, and then worked on all the details. Reviewers, I apologize for not having that foresight...

rainest commented:
Added 2972edf699 as the use of inline code in section headers did not play well with the sidebar:

![annotations](https://user-images.githubusercontent.com/571832/99012862-18235e00-2504-11eb-8587-555384497efe.png)

Otherwise spot checking around for formatting and nav looks fine. The Markdown quotes we used for marking required/optional stuff in deployment.md (e.g. https://deploy-preview-2446--kongdocs.netlify.app/kong-ingress-controller/1.0.x/concepts/deployment/#ingress-controller-deployment) don't look great, but they didn't look that great originally either. Not sure of a great way to handle those, and it's pretty minor, so may as well leave that as-is for now.

Looks generally fine aside from the known issues with the compatibility tables. IMO we do want them here; does the design team provide a set of standard icons we can pull from? I'd guess they have basic stuff for yes/no symbols we can use if so. If not, dunno, Twemoji? CSS stuff to shade the boxes?

lena-larionova commented:
> IMO we do want them here; does the design team provide a set of standard icons we can pull from? I'd guess they have basic stuff for yes/no symbols we can use if so. If not, dunno, Twemoji? CSS stuff to shade the boxes?

I've used Font Awesome in the past, though for now that means using
html tags (eg here https://docs.konghq.com/hub/plugins/compatibility/). I'll figure out how to make them more user-friendly at some point, but for now, that's probably what we'd go with.

lena-larionova commented:
Latest round of edits, hopefully the last for a minimal publish-ready doc:
* Added full KIC nav to CE and EE docs as interim solution - single-source CE/EE doc is the goal for next year, so this won't be an issue then.
* KIC now picks up the EE nav (Shane and Reza's decision) - it needed to pick up one of the navs for better flow, and the product decision was that EE should be it for exposure's sake. 
  * Maybe we need to add a banner or something across the KIC doc to assure users that it applies to both CE and EE?
* Fixed the icons on the version and plugin compatibility topics

@rainest @mflendrich I'm hoping this is pretty much it, fingers crossed. 

For Github logistics: I want to squash this PR because there's a mess of my commits in here, but I'm not sure this is the best idea for keeping the history coming from the KIC repo. Any advice?

rainest commented:
Icons look good. Nav is pretty much stuck with something imperfect due to the site structure; I don't think there's any great option there and we'll probably just need to answer questions as they come in until future reorganization.

`git rebase -i master` from CLI to squash yours as desired and `git push -f origin doc/kic-documentation` after to rewrite which parts of the history you'd like while keeping the rest. It will cancel any existing approval reviews, so someone will need to click past that again after.

tlblessing reviewed (approved):
Needs a copy edit as we discussed.
